Mathlib/AlgebraicGeometry/Stalks.lean

```diff
@@ -0,0 +1,245 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+
+! This file was ported from Lean 3 source module algebraic_geometry.stalks
+! leanprover-community/mathlib commit d39590fc8728fbf6743249802486f8c91ffe07bc
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.AlgebraicGeometry.PresheafedSpace
+import Mathlib.CategoryTheory.Limits.Final
+import Mathlib.Topology.Sheaves.Stalks
+
+/-!
+# Stalks for presheaved spaces
+
+This file lifts constructions of stalks and pushforwards of stalks to work with
+the category of presheafed spaces. Additionally, we prove that restriction of
+presheafed spaces does not change the stalks.
+-/
+
+
+noncomputable section
+
+universe v u v' u'
+
+open CategoryTheory
+
+open CategoryTheory.Limits CategoryTheory.Category CategoryTheory.Functor
+
+open AlgebraicGeometry
+
+open TopologicalSpace
+
+open Opposite
+
+variable {C : Type u} [Category.{v} C] [HasColimits C]
+
+attribute [local aesop safe cases (rule_sets [CategoryTheory])] Opens
```

Comment:
Oh, but you deleted the commented out `op_induction`! We'll want to restore that functionality, too.

Reply:
Ah yes, let me restore it

Reply:
Created an issue for this: #4551